src/custom-elements/parts-kit.tsx

```diff
@@ -5,6 +5,19 @@ export default class PartsKit extends HTMLElement {
   constructor() {
     // Always call super first in constructor
     super()
+
+    // Setup localStorage for theme prior to app loading to prevent FOUC
+    if (
+      localStorage.theme === 'dark' ||
+      (!('theme' in localStorage) &&
+        window.matchMedia('(prefers-color-scheme: dark)').matches)
+    ) {
+      localStorage.theme = 'dark'
+      document.documentElement.classList.add('dark')
+    } else {
+      localStorage.theme = 'light'
+      document.documentElement.classList.remove('dark')
+    }
```

Comment:
Not sure if this is the best place to have this, but I essentially want it to run in the `<head>` or before any content is rendered on the page. I figure placing it before the `React.render` was the way to go, but I'm open to other options if you guys have any opinions.

@joshuapease I also played around with creating a theme store, but it just ended up looking like more worth than was ultimately necessary. In my various testing, this has worked well.

Reply:
Works for me! If we ever feel the need to refactor we can.

Once this is deployed to the CDN, the CSS technically lives in the JS file, so there might not be much difference between adding the class here vs. within the React app. 

But... this route would let us manually set a background color or something if we really wanted to reduce or avoid FOUC (or whatever it's called).

Out of curiosity, does `dark` have to be applied on the `documentElement` or does it just need to be on a tag?

Reply:
It can be added to any parent tag, so we could have it applied on `app` and it'd still propagate down correctly.